File: server/src/main/java/com/seoulawake/backend/module/cafe/domain/Cafe.java
package com.seoulawake.backend.module.cafe.domain;

import com.seoulawake.backend.global.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Cafe extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "cafe_id")
    private Long id;
    private String name;
    private boolean isChecked;

    @Embedded
    private Location location;

    @Builder
    public Cafe(String name, Location location) {
        this.name = name;
        this.isChecked = false;
        this.location = location;
    }

    public void checked() {
        this.isChecked = true;
    }
}

File: backend/src/main/kotlin/seoulawake/server/module/cafe/domain/Cafe.kt
package seoulawake.server.module.cafe.domain

import jakarta.persistence.AttributeOverride
import jakarta.persistence.Column
import jakarta.persistence.Embedded
import jakarta.persistence.Entity
import seoulawake.server.global.common.BaseEntity
import seoulawake.server.global.exception.ErrorCode.PARAMETER_VIOLATION_ERROR
import seoulawake.server.global.exception.validate

@Entity
@AttributeOverride(name = "id", column = Column(name = "cafe_id"))
class Cafe(
  var name: String,

  @Embedded
  var address: Address,

  @Embedded
  var coordinates: Coordinates,

  @Embedded
  val status: Status = Status()
) : BaseEntity() {
  init {
    validate(this.name.isNotBlank()) { PARAMETER_VIOLATION_ERROR }
  }
}
